polygeist/lib/Dialect/Polygeist/IR/PolygeistOps.cpp

```diff
@@ -1468,14 +1468,15 @@ class MetaPointer2Memref final : public OpRewritePattern<Op> {
         return failure();
 
     Value val = src.getSource();
-    if (cast<LLVM::LLVMPointerType>(val.getType()).getElementType() !=
-        mt.getElementType())
-      val = rewriter.create<LLVM::BitcastOp>(
-          op.getLoc(),
-          LLVM::LLVMPointerType::get(
-              mt.getElementType(),
-              cast<LLVM::LLVMPointerType>(val.getType()).getAddressSpace()),
-          val);
+    if (auto PtrTy = dyn_cast<LLVM::LLVMPointerType>(val.getType())) {
```

Comment:
since `val` is the source of Pointer2MemrefOp, it must be `LLVMPointerType`
```suggestion
    auto PtrTy = cast<LLVM::LLVMPointerType>(val.getType()));
```

Reply:
Good point, adapted your suggestion.